### DISCUSSION TITLE
[Curso: Fundamentos do Blazor com .NET 8 - 3002] Autorização usando Roles

### DISCUSSION THREAD

**Discussion by thiagocajadev** [balta-io/forum#428] [Tira Dúvidas] [answered]
Olá Balta e pessoal. Apenas para referência, consegui habilitar a opção de autorização usando Roles.

No arquivo de inicialização, é preciso adicionar o **AddRoles** no **builder.Services.AddIdentityCore**, seguindo a sequencia abaixo:

```csharp
// Program.cs

builder.Services.AddIdentityCore<ApplicationUser>(options
    => options.SignIn.RequireConfirmedAccount = true)
    .AddRoles<IdentityRole>() // Adicione essa configuração
    .AddEntityFrameworkStores<ApplicationDbContext>()
    .AddSignInManager()
    .AddDefaultTokenProviders();
```    
Configurando a pagina Categories -> Delete.razor

```csharp
@page "/categories/delete/{id:int}"
@inject ApplicationDbContext Context
@inject NavigationManager Navigation
@rendermode InteractiveServer
@attribute [Authorize(Roles = "admin")] // habilitado atributo
```

Testando a pagina Auth.razor.

![auth-ok](https://github.com/balta-io/forum/assets/51033018/1b77e89c-ac87-480f-a87d-b214db6bedd4)

Após fazer logout, tentando excluir uma categoria.

![excluindo-sem-logar](https://github.com/balta-io/forum/assets/51033018/7de9a61b-cd68-44e5-9aa2-eddd26720411)

Redirecionado para fazer login conforme esperado.

![excluindo-sem-logar-redireciona](https://github.com/balta-io/forum/assets/51033018/d5d4e6cf-2471-47bf-a20e-0c31912f98cc)

Espero ajudar. Bons estudos e bons códigos! 👍

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @thiagokj
Uowwwwww, muito obrigado mesmo... isto vai salvar horas de pesquisa minha para os próximos conteúdos!!
Você é fera 🚀
